src/seedu/addressbook/parser/Parser.java

```diff
@@ -70,8 +77,17 @@ public Command parseCommand(String userInput) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
         }
 
-        final String commandWord = matcher.group("commandWord");
-        final String arguments = matcher.group("arguments");
+        String commandWord = matcher.group("commandWord");
+        String arguments = matcher.group("arguments");
+
+        if (!commandWord.equals(RepeatCommand.COMMAND_WORD)) {
```

Comment:
You have already used a switch case. Why not pass the command as an argument for RepeatCommand constructor?

Reply:
I think it would be good to pass the 'commandWord' and 'arguments' values as arguments for your RepeatCommand so that you can abstract this piece of code.

Reply:
Yup, it isn't parser's job to remember last command, can definitely use a better solution.